eth_keys/utils/asn1.py

```diff
@@ -0,0 +1,112 @@
+# Non-recoverable signatures are encoded using an asn1 sequence of two integers
+# We locally implement serialization and deserialization for this specific spec
+#   with constrained inputs.
+# This is done locally to avoid importing a 3rd-party library, in this very sensitive project.
+# asn1tools and pyasn1 were used as reference APIs, see how in tests/core/test_utils_asn1.py
```

Comment:
Maybe add a comment here that this module is for internal use only as it doesn't fully validate inputs (which is fine for us as we only use it to bridge to coincurve, but if someone uses it with unvalidated external inputs they might end up accepting invalid signatures).